### PR TITLE
fix: properly filter apis from admin context

### PR DESCRIPTION
**Pull request by guillaumelamirand**
## Issue

https://gravitee.atlassian.net/browse/CJ-778
https://gravitee.atlassian.net/browse/CJ-779
https://gravitee.atlassian.net/browse/CJ-780

## Description

When retrieving analytics as an admin, the apis wasn't properly filtered and then all apis analytics for all orga/env was retrieved.

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/EnvironmentAnalyticsResource.java [hunk 1]
```diff
@@ -232,31 +231,36 @@ public String field() {
...
-            fieldName = APPLICATION_FIELD;
-            ids =
-                applicationService
-                    .findIdsByUser(executionContext, getAuthenticatedUser())
-                    .stream()
-                    .filter(appId -> permissionService.hasPermission(executionContext, APPLICATION_ANALYTICS, appId, READ))
-                    .collect(Collectors.toList());
+            if (APPLICATION_FIELD.equalsIgnoreCase(analyticsParam.getField())) {
+                fieldName = APPLICATION_FIELD;
+                ids = applicationService.findIdsByOrganization(executionContext.getOrganizationId());
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/EnvironmentAnalyticsResource.java]: Shouldn't we look for applications in the current environment instead?

Comment (guillaumelamirand) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/EnvironmentAnalyticsResource.java]: Didn't think about it. Is an application scoped by env ?

Comment (phiz71) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/EnvironmentAnalyticsResource.java]: yes

Comment (guillaumelamirand) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/EnvironmentAnalyticsResource.java]: I have updated the code to use searchIds methods and use the same ApplicationQuery than findIdsByUser.

**Review by phiz71 — APPROVED** [review]

**Pull request merged**